clients/python/src/model_registry/integrator.py

```diff
@@ -0,0 +1,180 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+import re
+from typing import Any, Dict
+from .exceptions import StoreException
+from warnings import warn
+
+
+class supportedPlatform(Enum):
+    HUGGING_FACE = "HuggingFace"
+    MLFLOW = "MlFlow"
+
+    def __contains__(cls, item: str):
+        try:
+            cls(item)
+        except ValueError:
+            return False
+        return True
+
+
+class ModelInfoProvider(ABC):
+    @property
+    def passByInfo(self):
+        return ["version", "model_format_name", "model_format_version", "description"]
+
+    @property
+    def modelInfo(self):
+        raise NotImplementedError("This is an abstract class")
+
+    @abstractmethod
+    def get_model_info(self) -> dict:
+        pass
+
+    def _extractDirectPassByInfo(self, kwargs: dict) -> dict:
+        for keyword in self.passByInfo:
+            self._modelInfo[keyword] = kwargs[keyword]
+            del kwargs[keyword]
+        return kwargs
+
+
+class HuggingFaceModelInfoProvider(ModelInfoProvider):
+    _modelInfo = {}
+
+    def get_model_info(self, **kwargs) -> dict:
+        kwargs = self._extractDirectPassByInfo(kwargs)
+        return self._getHfModelInfo(**kwargs)
+
+    def _getHfModelInfo(
+        self, repo: str, path: str, *, author: str = None, model_name: str = None, git_ref: str = "main"
+    ) -> dict:
+        try:
+            from huggingface_hub import HfApi, hf_hub_url, utils
+        except ImportError as e:
+            msg = "huggingface_hub is not installed"
+            raise StoreException(msg) from e
+
+        api = HfApi()
+        try:
+            model_info = api.model_info(repo, revision=git_ref)
+        except utils.RepositoryNotFoundError as e:
+            msg = f"Repository {repo} does not exist"
+            raise StoreException(msg) from e
+
+        if not author:
+            author = model_info.author or "unknown"
```

Comment:
I see you dropped an except branch and a warning here, what's your rationale?

I understand this section might be a bit opinionated, so I'd like to hear you first.
<!--
I'd say for a high-level client these are important to make sure we communicate effectively with end users, to avoid e.g. meaningless issues because of poor usage, so it's common to wrap third-party errors if they actually might happen.
-->